packages/contractkit/src/wallets/signers/rpc-signer.ts

```diff
@@ -0,0 +1,130 @@
+import { ensureLeading0x, normalizeAddressWith0x, trimLeading0x } from '@celo/utils/src/address'
+import BigNumber from 'bignumber.js'
+import BN from 'bn.js'
+import { EncodedTransaction, Tx } from 'web3-core'
+import { RpcCaller } from '../../utils/rpc-caller'
+import { decodeSig } from '../../utils/signing-utils'
+import { Signer } from './signer'
+
+const currentTimeInSeconds = () => Math.round(Date.now() / 1000)
+
+const toRpcHex = (val: string | number | BigNumber | BN | undefined) => {
+  if (typeof val === 'number' || val instanceof BigNumber) {
+    return ensureLeading0x(val.toString(16))
+  } else if (typeof val === 'string') {
+    return ensureLeading0x(val)
+  } else {
+    return '0x0'
+  }
+}
+
+// TODO(yorke): move this into rpc-caller and generate typings from RPC spec
+enum RpcSignerEndpoint {
+  ImportAccount = 'personal_importRawKey',
+  UnlockAccount = 'personal_unlockAccount',
+  SignTransaction = 'eth_signTransaction',
+  SignBytes = 'eth_sign',
+}
+
+// tslint:disable-next-line: interface-over-type-literal
+type RpcSignerEndpointInputs = {
+  personal_importRawKey: [string, string]
+  personal_unlockAccount: [string, string, number]
+  eth_signTransaction: [any] // RpcTx doesn't match Tx because of nonce as string instead of number
+  eth_sign: [string, string]
+}
+
+// tslint:disable-next-line: interface-over-type-literal
+type RpcSignerEndpointResult = {
+  personal_importRawKey: string
+  personal_unlockAccount: boolean
+  eth_signTransaction: EncodedTransaction
+  eth_sign: string
+}
+
+/**
+ * Implements the signer interface on top of the JSON-RPC interface.
+ */
+export class RpcSigner implements Signer {
+  /**
+   * Construct a new instance of the RPC signer
+   *
+   * @param rpc RPC caller instance
+   * @param account Account address derived from the private key to be called in init
+   * @param unlockBufferSeconds Number of seconds to shrink the unlocked duration by to account for
+   * latency and timing inconsistencies on the node
+   * @param unlockTime Timestamp in seconds when the signer was last unlocked
+   * @param unlockDuration Number of seconds that the signer was last unlocked for
+   *
+   */
+  constructor(
+    protected rpc: RpcCaller,
+    protected account: string,
+    protected unlockBufferSeconds = 5,
+    protected unlockTime = -1,
```

Comment:
extracted because is it easy for testing? Otherwise could be more of an internal variable instead of something set from the dev

Reply:
I think `unlockTime` should be internal